### PR TITLE
openfortivpn-webview: init at 1.2.0

### PR DESCRIPTION
**Pull request by lriesebos**
## Description of changes

first contribution: added openfortivpn-webview package.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/op/openfortivpn-webview/package.nix
Comment (aos) [inline pkgs/by-name/op/openfortivpn-webview/package.nix:15]: Since this is a release on Github, you should use the `fetchFromGithub` fetcher instead: https://nix.dev/tutorials/packaging-existing-software.html#fetching-source-from-github

**Review by aos — COMMENTED** [review]
1. Use the `fetchFromGithub` fetcher
2. Make sure you follow commit conventions for contributing to nixpkgs: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions, specifically: rename your PR to follow the guideline above, split out the change to add yourself as the maintainer.

**Comment by lriesebos** [comment]
@aos appreciate the feedback for a first-time contributor. sorry for the delay too, just life happened. hope this PR is in a better state now!

**Review by aos — COMMENTED** [review]
Making some good progress

**Comment by lriesebos** [comment]
@aos , thanks again for the feedback, highly appreciated. let me know if there is anything else!

**Review by aos — COMMENTED** [review]
### Edit: Below is incorrect. See: https://github.com/NixOS/rfcs/pull/140, but will keep below for posterity.
<hr />
1. Rename from `package.nix` to `default.nix`
2. You should add the package to `pkgs/top-level/all-packages.nix`, alphabetically.

After these changes, you should be able to build it from the root of the repo with:
```
nix-build -A openfortivpn-webview
```
Once you do - test it out and make sure it works as you expect (perhaps including any version information).

I think that should be everything remaining from my point of view.

**Comment by lriesebos** [comment]
@aos , I am a bit confused. the [quick start to adding a package](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#quick-start-to-adding-a-package) specifically asks for `package.nix`. also, there is no mention of `all-packages.nix` for adding a new package, and the new package builds fine as is. see the output below. recent PR's also seem to point in the same direction (e.g. https://github.com/NixOS/nixpkgs/pull/319908). are you sure your latest change requests are really supposed to be there?

```
leon@tokachi:~/code/nixpkgs$ nix-build -A openfortivpn-webview
/nix/store/9d0ssvxi5ins6as1c1x5r81pkcgp8b5l-openfortivpn-webview-1.2.0
leon@tokachi:~/code/nixpkgs$ nix build .#openfortivpn-webview
leon@tokachi:~/code/nixpkgs$ ls -l result
lrwxrwxrwx. 1 leon leon 70 Jun 15 23:08 result -> /nix/store/9d0ssvxi5ins6as1c1x5r81pkcgp8b5l-openfortivpn-webview-1.2.0
leon@tokachi:~/code/nixpkgs$ ./result/bin/openfortivpn-webview --help
openfortivpn-webview [command]

Commands:
  openfortivpn-webview [host:port]  Connect to the given host

Options:
  --version         Show version number                                [boolean]
  --realm           The authentication realm.                           [string]
  --url             The already built SAML URL. This takes precedence over
                    [host:port].                                        [string]
  --url-regex       A regex to detect the URL that needs to be visited before
                    printing SVPNCOOKIE.
                                      [string] [default: "/sslvpn/portal\.html"]
  --keep-open       Do not close the browser automatically.
  --proxy-server    HTTP Proxy in the format hostname:port.             [string]
  --extra-ca-certs  Path to a file with extra certificates. The file should
                    consist of one or more trusted certificates in PEM format.
                                                                        [string]
  --trusted-cert    The fingerprint of a certificate to always trust, even if
                    invalid. The details of invalid certificates, fingerprint
                    included, will be dumped in the console.            [string]
  --help            Show help                                          [boolean]
```

**Review by aos — APPROVED** [review]
@lriesebos Ah thanks for the correction. Looks like [RFC 140](https://github.com/NixOS/rfcs/pull/140) was merged somewhat recently and the guidelines have also been updated. In that case - disregard my last comment. This is good to go from my point of view :-)

**Comment by lriesebos** [comment]
if there is anything else that needs to be done to unblock this, let me know

**Comment by aos** [comment]
@lriesebos I would advertise this PR here: https://discourse.nixos.org/t/prs-already-reviewed/2617 or https://discourse.nixos.org/t/prs-ready-for-review-december/1711

File: pkgs/by-name/op/openfortivpn-webview/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/op/openfortivpn-webview/package.nix:14]:
```suggestion
in
buildNpmPackage {
  pname = "openfortivpn-webview";
  inherit version;
```

Comment (lriesebos) [reply pkgs/by-name/op/openfortivpn-webview/package.nix]: thanks for the suggestion, updated that accordingly! also noticed that RFC-166 formatting came through in the meanwhile, so got that reformatted too (after two retries).

**Comment by lriesebos** [comment]
@SuperSandro2000 I think I addressed your last concerns. let me know if there is anything else I can do!

**Comment by SuperSandro2000** [comment]
@ofborg build openfortivpn-webview

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**
